go/enclave/components/rollup_consumer.go

```diff
@@ -50,9 +50,55 @@ func NewRollupConsumer(
 	}
 }
 
+func (rc *rollupConsumerImpl) ProcessRollupData(ctx context.Context, processed *common.ProcessedL1Data) ([]common.ExtRollupMetadata, error) {
+	rollupTxs := processed.GetEvents(common.RollupTx)
+	rollups := make([]*common.ExtRollup, 0, len(rollupTxs))
+	txsSeen := make(map[gethcommon.Hash]bool)
+
+	for _, rtx := range rollupTxs {
+		// verify signature and decode rollups
```

Comment:
in my mind, the content of this loop was the main building block in the "ProcessRollupData" function. Basically, it would be responsible for a single rollup, and the loop logic would be in the layer above